### DISCUSSION TITLE
rectSwappingStrategy Grid Ordering on drag end

### DISCUSSION THREAD

**Discussion by joeflem** [clauderic/dnd-kit#485] [Q&A]
Hi there 👋

I'm struggling a little bit with the ordering of a grid using rectSwappingStrategy. It's probably something we've done wrong when setting up this package though.

https://codesandbox.io/s/react-dnd-kit-swapping-order-sp27y

In the example above, if you shrink down the preview so you have two columns, dragging tile 4 to tile 1 looks okay whilst you're dragging, but on release it moves tile 1 to tile 2, which isn't the expected behaviour when using rectSwappingStrategy. At first I wondered if it was because we are using an array of objects as mentioned in #318, however our objects do contain an ID property so should be okay.

Has anyone else had this behaviour and found a fix? 

https://user-images.githubusercontent.com/45002285/138670687-fde106d2-091e-4d17-a547-2afe452c6e46.mov

Thanks in advance!

**Comment by psd8** [comment]
@joeflem In the latest version, they have introduced a new method called **arraySwap** which works superbly! I was stuck on the same issue and just found the solution with this method.

Here is a code sandbox example
https://codesandbox.io/s/react-dnd-kit-swapping-order-forked-m2pdlw?file=/src/Grid/Grid.jsx
